ui/src/notebooks/context/notebook.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {FC, useState} from 'react'
+
+// TODO make this polymorphic to mimic the self registration
+// of pipe stages
+export type Pipe = any
+
+export interface NotebookContextType {
+  id: string
+  pipes: Pipe[]
+  addPipe: (pipe: Pipe) => void
+  updatePipe: (idx: number, pipe: Pipe) => void
+  removePipe: (idx: number) => void
+}
+
+export const DEFAULT_CONTEXT = {
+  id: 'new',
+  pipes: [],
+  addPipe: () => {},
+  updatePipe: () => {},
+  removePipe: () => {},
+}
+
+// NOTE: this just loads some test data for exploring the
+// interactions between types. Make sure it's never true
+// during the pull review process
+const TEST_MODE = false
```

Comment:
lets consider hiding this behind a feature flag

Reply:
not sure why we would, but maybe i'm misunderstanding the intent here.